planner/core/logical_plan_builder.go

```diff
@@ -603,19 +602,18 @@ func (b *PlanBuilder) buildSelection(ctx context.Context, p LogicalPlan, where a
 
 // buildProjectionFieldNameFromColumns builds the field name, table name and database name when field expression is a column reference.
 func (b *PlanBuilder) buildProjectionFieldNameFromColumns(origField *ast.SelectField, colNameField *ast.ColumnNameExpr, c *expression.Column) (colName, origColName, tblName, origTblName, dbName model.CIStr) {
-	origColName, tblName, dbName = colNameField.Name.Name, colNameField.Name.Table, colNameField.Name.Schema
-	if origField.AsName.L != "" {
-		colName = origField.AsName
+	origTblName, origColName, dbName = c.OrigTblName, c.OrigColName, c.DBName
+	if origField.AsName.L == "" {
+		colName = colNameField.Name.Name
 	} else {
-		colName = origColName
+		colName = origField.AsName
 	}
 	if tblName.L == "" {
 		tblName = c.TblName
+	} else {
+		tblName = colNameField.Name.Table
 	}
-	if dbName.L == "" {
-		dbName = c.DBName
-	}
-	return colName, origColName, tblName, c.OrigTblName, c.DBName
+	return
```

Comment:
This `return` can be removed.

Reply:
In Golang, if function has return values, it **must** explicit call `return` keyword. e.g
```golang
func f1() {
	fmt.Println("f1")
}
// print: f1
```
vs
```
func f1() (value int){
	fmt.Println("f2")
	value = 1
	return
}
// compiler error: missing return at end of function
```